File: Implementation/C#/Source/Text/Reader/Model/VEntry.cs
using StringBuilder = System.Text.StringBuilder;
using TextWriter = System.IO.TextWriter;

namespace Cks.Text.Reader.Model
{

public sealed class VEntry : Value
{
	public readonly string Name;
	public readonly Value Value;

	public VEntry(SourcePos SourcePos, string Name, Value Value) : base(SourcePos)
	{
		this.Name = Name;
		this.Value = Value;
	}

	public override void ToString(StringBuilder buf)
	{
		buf.Append(Name);
		buf.Append(": ");
		Value.ToString(buf);
	}

	protected override string DumpHeader()
	{
		return Name + ":";
	}

	protected override void DumpChildren(TextWriter Out, uint Depth)
	{
		Value.Dump(Out, Depth);
	}
}

}

File: Implementation/Java/Source/surface/model/VEntry.java
package cks.surface.model;

import cks.io.SourcePos;

import java.io.PrintWriter;

public class VEntry extends Value
{
	public final String name;
	public final Value value;

	public VEntry(SourcePos loc, String name, Value value)
	{
		super(loc);
		this.name = name;
		this.value = value;
	}

	public void toString(StringBuilder b)
	{
		super.toString(b);
		b.append(": ");
		b.append(this.name);
	}

	protected void dumpChildren(PrintWriter out, int depth)
	{
		this.value.dump(out, depth);
		super.dumpChildren(out, depth);
	}
}
